storage/blockdevice/tests/UNITTESTS/blockdevice/FlashSimBlockDevice/CMakeLists.txt

```diff
@@ -0,0 +1,24 @@
+# Copyright (c) 2021 ARM Limited. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+set(TEST_NAME flash-sim-blockdevice-unittest)
+
+add_executable(${TEST_NAME})
+
+target_sources(${TEST_NAME}
+    PRIVATE
+        ${mbed-os_SOURCE_DIR}/storage/blockdevice/source/FlashSimBlockDevice.cpp
+        test_FlashSimBlockDevice.cpp
+)
+
+target_link_libraries(${TEST_NAME}
+    PRIVATE
+        mbed-headers
+        mbed-stubs-headers
+        mbed-stubs-platform      
```

Comment:
`mbed-stubs-platform` is target_link_libraries with PRIVATE against `mbed-stubs-headers` and it will not visible in FlashSimBlockDevice CMake and based on our previous discussion here https://github.com/ARMmbed/mbed-os/pull/14285#discussion_r576287699 decided to go with PRIVATE

Reply:
I think we could've avoided this circular flow of dependencies if we just linked to the header libraries being used in the stubs instead of adding "all available headers" everywhere, as I mentioned in another comment. This is probably ok for now though.